v-next/hardhat/src/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -307,12 +292,16 @@ export class NetworkManagerImplementation implements NetworkManager {
    * @returns a valid network configuration including any config additions from
    *   plugins
    */
-  async #resolveNetworkConfig(
+  async #resolveNetworkConfig<ChainTypeT extends ChainType | string>(
     resolvedNetworkName: string,
     networkConfigOverride: NetworkConfigOverride | undefined = {},
+    resolvedChainType: ChainTypeT,
   ): Promise<NetworkConfig> {
     const existingNetworkConfig = this.#networkConfigs[resolvedNetworkName];
-    if (Object.keys(networkConfigOverride).length === 0) {
+    if (
+      Object.keys(networkConfigOverride).length === 0 &&
+      resolvedChainType === existingNetworkConfig.chainType
+    ) {
```

Comment:
Can I check I have followed the change here.
In the previous commit it only checked `networkConfigOverride` was empty i.e. no overrides. This was wrong - because the chainType is now provided as well as the overrides. So this check need to be updated, it should only short circuit if there are no overrides and the chainType remains the same as the original network config.